cli/command/task.go

```diff
@@ -10,6 +16,221 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
+						Name:  "env",
+						Usage: "environment variable override in format 'CONTAINER:VAR=VAL' (can be specified multiple times)",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a Task",
+				ArgsUsage: "NAME",
+				Action:    t.delete,
+			},
+			{
+				Name:      "list",
+				Usage:     "list all Tasks",
+				Action:    t.list,
+				ArgsUsage: " ",
+			},
+			{
+				Name:      "read",
+				Usage:     "describe a Task",
+				Action:    t.read,
+				ArgsUsage: "NAME",
+			},
+			{
+				Name:      "logs",
+				Usage:     "get the logs for a task",
+				Action:    t.logs,
+				ArgsUsage: "NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "DEPLOY", "NAME")
+	if err != nil {
+		return err
+	}
+
+	overrides, err := parseOverrides(c.StringSlice("env"))
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
+		TaskName:           args["NAME"],
+		EnvironmentID:      environmentID,
+		DeployID:           deployID,
+		ContainerOverrides: overrides,
+	}
+
+	jobID, err := t.client.CreateTask(req)
+	if err != nil {
+		return err
+	}
+
+	return t.waitOnJobHelper(c, jobID, "creating", func(taskID string) error {
+		task, err := t.client.ReadTask(taskID)
+		if err != nil {
+			return err
+		}
+
+		return t.printer.PrintTasks(task)
+	})
+}
+
+func (t *TaskCommand) delete(c *cli.Context) error {
+	return t.deleteHelper(c, "task", func(taskID string) (string, error) {
+		return t.client.DeleteTask(taskID)
+	})
+}
+
+func (t *TaskCommand) list(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintTaskSummaries(taskSummaries...)
+}
+
+func (t *TaskCommand) read(c *cli.Context) error {
+
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	taskExists := func(id string) bool {
+		for _, t := range taskSummaries {
+			if t.TaskID == id {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	taskIDs, err := t.resolver.Resolve("task", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	tasks := make([]*models.Task, len(taskIDs))
+	for i, taskID := range taskIDs {
+		if taskExists(taskID) {
```

Comment:
we should throw in a debug statement here: 
```
if !taskExists[taskID] {
    log.Printf("[DEBUG] Resolver returned an expired task '%s'", taskID)
    continue
}

task, err := t.client.ReadTask(taskID)
...
```

(note that we try to avoid if/else blocks by using `continue` statements)

Reply:
Does it need the continue here? Won't it continue without it?